avahi-daemon/static-services.c

```diff
@@ -403,8 +412,29 @@ static void XMLCALL xml_start(void *data, const char *el, const char *attr[]) {
 
         u->current_tag = XML_TAG_PORT;
     } else if (u->current_tag == XML_TAG_SERVICE && strcmp(el, "txt-record") == 0) {
-        if (attr[0])
-            goto invalid_attr;
+        if (attr[0]) {
+            if (strcmp(attr[0], "value-format") == 0) {
+                txt_record_value_type value_type;
+
+                if (strcmp(attr[1], "text") == 0) {
```

Comment:
I’m not familiar with this XML parsing code, but is it possible for `attr[1]` to be `NULL` here? If so, `strcmp()` will explode.